### PR TITLE
tests: benchmarks: multicore: Start testing IPC with s2ram

### PR DESCRIPTION
**Pull request by nordic-bami**
Run IPC benchmark with s2ram

### CODE REVIEW

File: tests/benchmarks/multicore/idle_ipc/src/main.c [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+		printf("ipc_service_register_endpoint() failure (%d)", ret);
+		return ret;
+	}
+
+	if (IS_ENABLED(CONFIG_PM_DEVICE_RUNTIME)) {
+		pm_device_runtime_enable(console_dev);
+		pm_device_runtime_enable(ipc0_instance);
+	}
+
+	k_sem_take(&bound_sem, K_FOREVER);
```
Comment (nordic-piks) [inline tests/benchmarks/multicore/idle_ipc/src/main.c:107]: Maybe some specific timeout and assert when bound did not happen?

File: tests/benchmarks/multicore/idle_ipc/src/main.c [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+	const struct device *ipc0_instance;
+	struct ipc_ept ep;
+	int ret;
+	unsigned long last_cnt = 0;
+	unsigned long delta = 0;
+
+	p_payload = (struct payload *)k_malloc(CONFIG_APP_IPC_SERVICE_MESSAGE_LEN);
+	if (!p_payload) {
+		printk("k_malloc() failure\n");
+		return -ENOMEM;
```
Comment (nordic-piks) [inline tests/benchmarks/multicore/idle_ipc/src/main.c:78]: please do assert here, as exit from main will lead to s2ram, thus it is hard to find what was the problem

File: tests/benchmarks/multicore/idle_ipc/src/main.c [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+	p_payload->cnt = 0;
+
+	printk("IPC-service %s demo started\n", CONFIG_BOARD_TARGET);
+
+	ipc0_instance = DEVICE_DT_GET(DT_NODELABEL(ipc0));
+
+	ret = ipc_service_open_instance(ipc0_instance);
+	if ((ret < 0) && (ret != -EALREADY)) {
+		LOG_INF("ipc_service_open_instance() failure (%d)", ret);
+		return ret;
```
Comment (nordic-piks) [inline tests/benchmarks/multicore/idle_ipc/src/main.c:93]: assert on any error

File: tests/benchmarks/multicore/idle_ipc/src/main.c [hunk 4]
```diff
@@ -0,0 +1,139 @@
...
+	}
+
+	ret = ipc_service_register_endpoint(ipc0_instance, &ep, &ep_cfg);
+	if (ret < 0) {
+		printf("ipc_service_register_endpoint() failure (%d)", ret);
+		return ret;
+	}
```
Comment (nordic-piks) [inline tests/benchmarks/multicore/idle_ipc/src/main.c:100]: assert on any error

File: tests/benchmarks/multicore/idle_ipc/prj.conf [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+CONFIG_PM_S2RAM=y
+CONFIG_PM_S2RAM_CUSTOM_MARKING=y
+CONFIG_PM_DEVICE=y
+CONFIG_POWEROFF=y
+CONFIG_PM_DEVICE_RUNTIME=y
+
+CONFIG_ASSERT=y
+
+# Enable for debugging pruposes only
+CONFIG_PRINTK=y
```
[the remark this answers is not in the document]
Comment (nordic-bami) [reply tests/benchmarks/multicore/idle_ipc/prj.conf:24]: No - to be fixed

**Review by nordic-piks — APPROVED** [review]

**Review by nordic-pikr — APPROVED** [review]

**Review by nordic-segl — APPROVED** [review]

File: tests/benchmarks/multicore/idle_ipc/src/main.c [hunk 6]
```diff
@@ -0,0 +1,144 @@
+/*
+ * Copyright (c) 2022 Nordic Semiconductor ASA
```
Comment (nordic-segl) [inline tests/benchmarks/multicore/idle_ipc/src/main.c:2]: :smile:

File: tests/benchmarks/multicore/idle_ipc/sysbuild.cmake [hunk 7]
```diff
@@ -0,0 +1,22 @@
+#
+# Copyright (c) 2023 Nordic Semiconductor ASA
```
Comment (nordic-segl) [inline tests/benchmarks/multicore/idle_ipc/sysbuild.cmake:2]: :smile:

**Review by nordic-segl — APPROVED** [review]

**Comment by nordic-bami** [comment]
Please review: @nrfconnect/ncs-co-build-system

File: tests/benchmarks/multicore/idle_ipc/Kconfig.sysbuild [hunk 8]
```diff
@@ -0,0 +1,10 @@
+#
+# Copyright (c) 2024 Nordic Semiconductor ASA
+#
+# SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+#
+
+source "${ZEPHYR_BASE}/share/sysbuild/Kconfig"
+
+config REMOTE_BOARD
```
Comment (nordicjm) [inline tests/benchmarks/multicore/idle_ipc/Kconfig.sysbuild:9]: Copy what was done here: https://github.com/nrfconnect/sdk-nrf/pull/19018/files#diff-5b9bb7529743d0bc971625c8f49d0e07dcc10b26ff5a56228cdb797c0711021cR9

File: tests/benchmarks/multicore/idle_ipc/sysbuild/nrf54h20dk_nrf54h20_cpurad.conf [hunk 9]
```diff
@@ -0,0 +1 @@
+SB_CONFIG_REMOTE_BOARD="nrf54h20dk/nrf54h20/cpurad"
```
Comment (nordicjm) [inline tests/benchmarks/multicore/idle_ipc/sysbuild/nrf54h20dk_nrf54h20_cpurad.conf:1]: file goes

File: tests/benchmarks/multicore/idle_ipc/testcase.yaml [hunk 10]
```diff
@@ -0,0 +1,31 @@
...
+        - "${CUSTOM_ROOT_TEST_DIR}/test_measure_power_consumption.py::test_measure_and_data_dump_power_consumption_ipc"
+
+  benchmarks.multicore.idle_ipc.shifted.nrf54h20dk_cpuapp_cpurad.s2ram:
+    harness: pytest
+    platform_allow:
+      - nrf54h20dk/nrf54h20/cpuapp
+    integration_platforms:
+      - nrf54h20dk/nrf54h20/cpuapp
+    extra_args:
+      - SB_CONF_FILE=sysbuild/nrf54h20dk_nrf54h20_cpurad.conf
```
Comment (nordicjm) [inline tests/benchmarks/multicore/idle_ipc/testcase.yaml:26]: line goes

File: tests/benchmarks/multicore/idle_ipc/testcase.yaml [hunk 11]
```diff
@@ -0,0 +1,31 @@
...
+
+tests:
+  benchmarks.multicore.idle_ipc.nrf54h20dk_cpuapp_cpurad.s2ram:
+    harness: pytest
+    platform_allow:
+      - nrf54h20dk/nrf54h20/cpuapp
+    integration_platforms:
+      - nrf54h20dk/nrf54h20/cpuapp
+    extra_args:
+      - SB_CONF_FILE=sysbuild/nrf54h20dk_nrf54h20_cpurad.conf
```
Comment (nordicjm) [inline tests/benchmarks/multicore/idle_ipc/testcase.yaml:13]: line goes

**Review by nordicjm — APPROVED** [review]

**Pull request merged**